distribution/pull_v2_windows.go

```diff
@@ -23,16 +23,17 @@ func (ld *v2LayerDescriptor) Descriptor() distribution.Descriptor {
 }
 
 func (ld *v2LayerDescriptor) open(ctx context.Context) (distribution.ReadSeekCloser, error) {
-	if len(ld.src.URLs) == 0 {
-		blobs := ld.repo.Blobs(ctx)
-		return blobs.Open(ctx, ld.digest)
-	}
-
 	var (
 		err error
 		rsc distribution.ReadSeekCloser
 	)
 
+	// Try to get the layer from the registry.
+	blobs := ld.repo.Blobs(ctx)
+	if rsc, err = blobs.Open(ctx, ld.digest); err == nil {
+		return rsc, err
```

Comment:
That will probably work. I like the idea of falling back to the foreign URL later on in the process if something goes wrong with the download instead of trying to make the decision entirely inside `open`, but looking at the code that seems a lot more complicated.